emcc.py

```diff
@@ -285,6 +287,64 @@ def will_metadce():
   return settings.OPT_LEVEL >= 3 or settings.SHRINK_LEVEL >= 1
 
 
+def create_reproduce_file(name, args):
+  def make_relative(filename):
+    filename = os.path.normpath(os.path.abspath(filename))
+    filename = os.path.splitdrive(filename)[1]
+    filename = filename[1:]
+    return filename
+
+  root = unsuffixed_basename(name)
+  with tarfile.open(name, 'w') as reproduce_file:
+    reproduce_file.add(shared.path_from_root('emscripten-version.txt'), os.path.join(root, 'version.txt'))
+
+    with shared.get_temp_files().get_file(suffix='.t') as rsp_name:
+      with open(rsp_name, 'w') as rsp:
+        ignore_next = False
+        output_arg = None
+
+        for arg in args:
+          if arg.startswith('--reproduce='):
+            continue
+
+          if arg.startswith('-o='):
+            rsp.write('-o\n')
+            arg = arg[3:]
+            output_arg = True
+            ignore_next = True
```

Comment:
If this arg is `-o=foo` why are we ignoring the next thing after it? The total args could be `-o=foo input.c` but we want to process `input.c`. I'm probably reading this wrong...

Reply:
Setting ignore_next at this point is more like saying "ignore current" since the handling of ignore_next has yet to happen. .  Its  little confusing maybe.. but it works.